packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -710,30 +724,37 @@ contract Deploy is Deployer {
 
     /// @notice Transfer ownership of the DisputeGameFactory contract to the final system owner
     function transferDisputeGameFactoryOwnership() onlyDevnet broadcast() public {
-        DisputeGameFactory disputeGameFactory = DisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
-        address owner = disputeGameFactory.owner();
-        address finalSystemOwner = cfg.finalSystemOwner();
-        if (owner != finalSystemOwner) {
-            disputeGameFactory.transferOwnership(finalSystemOwner);
-            console.log("DisputeGameFactory ownership transferred to: %s", finalSystemOwner);
+        string[2] memory contractNames = ["DisputeGameFactoryProxy", "MIPSDisputeGameFactoryProxy"];
+        for (uint256 i; i < contractNames.length; i++) {
+            DisputeGameFactory disputeGameFactory = DisputeGameFactory(mustGetAddress(contractNames[i]));
+            address owner = disputeGameFactory.owner();
+            address finalSystemOwner = cfg.finalSystemOwner();
+            if (owner != finalSystemOwner) {
+                disputeGameFactory.transferOwnership(finalSystemOwner);
+                console.log("%s ownership transferred to: %s", contractNames[i], finalSystemOwner);
+            }
         }
     }
 
     /// @notice Sets the implementation for the `FAULT` game type in the `DisputeGameFactory`
     function setFaultGameImplementation() onlyDevnet broadcast() public {
-        DisputeGameFactory factory = DisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
-        Claim absolutePrestate = Claim.wrap(bytes32(cfg.faultGameAbsolutePrestate()));
-        IBigStepper faultVm = IBigStepper(new AlphabetVM(absolutePrestate));
-        if (address(factory.gameImpls(GameTypes.FAULT)) == address(0)) {
-            factory.setImplementation(GameTypes.FAULT, new FaultDisputeGame({
-                _absolutePrestate: absolutePrestate,
-                _maxGameDepth: cfg.faultGameMaxDepth(),
-                _gameDuration: Duration.wrap(uint64(cfg.faultGameMaxDuration())),
-                _vm: faultVm,
-                _l2oo: L2OutputOracle(mustGetAddress("L2OutputOracleProxy")),
-                _blockOracle: BlockOracle(mustGetAddress("BlockOracle"))
-            }));
-            console.log("DisputeGameFactory: set `FaultDisputeGame` implementation");
+        string[2] memory contractNames = ["DisputeGameFactoryProxy", "MIPSDisputeGameFactoryProxy"];
+
+        for (uint256 i; i < contractNames.length; i++) {
+            DisputeGameFactory factory = DisputeGameFactory(mustGetAddress(contractNames[i]));
+            Claim absolutePrestate = Claim.wrap(bytes32(cfg.faultGameAbsolutePrestate()));
```

Comment:
Not necessarily a blocker for this PR, but we will need to have different absolute pre-states for the alphabet and cannon games.